### PR TITLE
Setting up system test with capybara

### PR DESCRIPTION
**Pull request by chelobotix**
In this PR we fulfilled the following tasks:
- [x] Resolve the N+1 problem when fetching all posts and their comments for a user
- [x] Use Capybara to write integration tests for each view in your project.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.

User show page:
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.

User post index page:
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.

Post-show page:
I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/system/post/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe 'Index posts page', type: :system do
```
Comment (julie-ify) [inline spec/system/post/post_spec.rb:3]: - [x] **[OPTIONAL]** I think file should be renamed `index_spec.rb` as that is more intuitive than `post_spec.rb`

File: spec/system/user/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe 'Index users page', type: :system do
```
Comment (julie-ify) [inline spec/system/user/users_spec.rb:3]: - [x] **[OPTIONAL]** I think file should be renamed `index_spec.rb` as that is more intuitive than `users_spec.rb`

File: spec/system/post/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                        likes_counter: 0,
+                        comments_counter: 0,
+                        author: user1)
+    Post.create(id: 2, title: 'Post Number 2', text: 'Lorem Ipsu', likes_counter: 0, comments_counter: 0, author: user1)
+    Post.create(id: 3, title: 'Post Number 3', text: 'Lorem Ipsu', likes_counter: 0, comments_counter: 0, author: user1)
+    Post.create(id: 4, title: 'Post Number 4', text: 'Lorem Ipsu', likes_counter: 0, comments_counter: 0, author: user1)
+    Comment.create(author: user1, post: post1, text: 'Awesome post!')
+    Like.create(author: user1, post: post1)
+  end
+  describe 'index page' do
```
Comment (julie-ify) [inline spec/system/post/show_spec.rb:23]: - [x] **[OPTIONAL]** You have a typo error here. Kindly fix it.

  ```
  describe 'show page' do
  ```

**Review by julie-ify — APPROVED** [review]
Hi @chelobotix and @luis-pomare,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/post/post_spec.rb (hunk 1 above)
Comment (chelobotix) [reply spec/system/post/post_spec.rb:3]: Excellent, I will rename that

File: spec/system/user/users_spec.rb (hunk 2 above)
Comment (chelobotix) [reply spec/system/user/users_spec.rb:3]: Excellent, I will rename that

File: spec/system/post/show_spec.rb (hunk 3 above)
Comment (chelobotix) [reply spec/system/post/show_spec.rb:23]: Excellent, I will fix it

**Pull request merged**